app/controllers/spree/admin/reports_controller_decorator.rb

```diff
@@ -181,20 +189,37 @@ def sales_tax
   end
 
   def bulk_coop
-    # -- Prepare date parameters
-    prepare_date_params params
+    if request.format.json?
```

Comment:
This would be too early to do that. Refactoring and cleanup will have to happen at some point anyway.
For now it is more clear what's happening in controller for other developers and it reduces controller duplication if all actions would need to be copy/pasted to API controller. Also Rails still support `respond_to`, meaning that controller can respond to different formats.